src/EventAccumulator.ts

```diff
@@ -0,0 +1,99 @@
+import type { CountlyWebSdk, IEventAccumulator, CountlyEvent, CountlyEventData } from 'countly-sdk-web'
+
+const eventDefaults: CountlyEventData = {
+  key: '',
+  count: 1,
+  sum: 1,
+  dur: Date.now(),
+  segmentation: {}
+}
+
+interface eventStore {
+  eventData: CountlyEventData
+  startTime: number
+  timeout: NodeJS.Timeout
+}
+
+/**
+ * EventAccumulator is a class that accumulates events and flushes them to the Countly server.
+ */
+export class EventAccumulator implements IEventAccumulator {
+  private readonly metricsService: CountlyWebSdk
+  private readonly events: Record<string, eventStore>
+  private readonly flushInterval: number
+
+  /**
+   * Create a new EventAccumulator
+   *
+   * @param {CountlyWebSdk} metricsService - instance
+   * @param {number} flushInterval - in milliseconds
+   */
+  constructor (metricsService: CountlyWebSdk, flushInterval: number = 5 * 60 * 1000) {
+    this.metricsService = metricsService
+    this.flushInterval = flushInterval
+    this.events = {}
```

Comment:
```suggestion
    this.events = new Map<string, eventStore>()
```

Reply:
I usually start with unknown and drill down from there... but I agree that using some object literals makes getting the code out there a little bit less of a burden :) Glad to hear you got it sorted